mat/dense_arithmetic.go

```diff
@@ -504,12 +504,12 @@ func (m *Dense) Exp(a Matrix) {
 	a1.Copy(a)
 	v := getWorkspace(r, r, true)
 	vraw := v.RawMatrix()
-	vvec := blas64.Vector{Inc: 1, Data: vraw.Data}
+	vvec := blas64.Vector{N: r * r, Inc: 1, Data: vraw.Data}
```

Comment:
This gets recalculated a lot; `n := r*r`? (though it did previously just differently).